### PR TITLE
AWS DataSource: Fix namespaces in sagemaker metrics

### PR DESCRIPTION
**Pull request by tristanburgess**
This is part of https://github.com/grafana/cloud-onboarding/issues/6236 and a continuation of https://github.com/grafana/grafana/pull/85009

**What is this feature?**

This fixes namespaces involved with some SageMaker metrics added in the above linked PR [SageMaker](https://docs.aws.amazon.com/sagemaker/latest/dg/monitoring-cloudwatch.html#cloudwatch-metrics-endpoint-invocation) for all namespaces except for GroundTruth and FeatureStore, which our customer did not seem to have a need for.

**Why do we need this feature?**

Per the above linked ticket, we have customers that are interested in several SageMaker namespaces that we do not currently have support for

**Who is this feature for?**

AWS SageMaker users for the CW Integration and/or DataSource

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tristanburgess** [comment]
not sure where

```
--- FAIL: TestGetRandomDigits (0.13s)
    encoding_test.go:218: 
        	Error Trace:	/drone/src/pkg/util/encoding_test.go:218
        	Error:      	"34.132859999999994" is not less than "33.7199"
        	Test:       	TestGetRandomDigits
        	Messages:   	Chi squared value must be less than the 99.99% critical bound
FAIL
coverage: 62.1% of statements
FAIL	github.com/grafana/grafana/pkg/util	0.471s
```

is coming from, but in any case, I don't seem to have permissions to restart this failed Drone test.

**Comment by tristanburgess** [comment]
merged main 🤞

File: pkg/tsdb/cloudwatch/constants/metrics.go [hunk 1]
```diff
@@ -449,27 +449,33 @@ var NamespaceMetricsMap = map[string][]string{
...
-	"AWS/SageMaker":            {"ClientInvocations", "ClientInvocationErrors", "ClientLatency", "CPUUtilization", "DatasetObjectsAutoAnnotated", "DatasetObjectsHumanAnnotated", "DatasetObjectsLabelingFailed", "DiskUtilization", "ExecutionStarted", "ExecutionFailed", "ExecutionSucceeded", "ExecutionStopped", "ExecutionDuration", "GPUMemoryUtilization", "GPUUtilization", "Invocation4XXErrors", "Invocation5XXErrors", "InvocationModelErrors", "Invocations", "InvocationsPerCopy", "InvocationsPerInstance", "JobsFailed", "JobsStopped", "JobsSucceeded", "LoadedModelCount", "MemoryUtilization", "ModelCacheHit", "ModelDownloadingTime", "ModelLatency", "ModelLoadingTime", "ModelLoadingWaitTime", "ModelSetupTime", "ModelUnloadingTime", "NumberOfUsers", "OverheadLatency", "StepStarted", "StepFailed", "StepSucceeded", "StepStopped", "StepDuration", "TotalDatasetObjectsLabeled"},
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/constants/metrics.go:452]: Can you copy paste the [old SageMaker metrics and dimensions](https://github.com/grafana/grafana/pull/85009/files) and dimensions from before the last pr to the new one? It seems like the [Ground Truth Metrics](https://docs.aws.amazon.com/sagemaker/latest/dg/monitoring-cloudwatch.html#cloudwatch-metrics-ground-truth) from the docs that were in the old SageMaker metrics are missing.

Comment (tristanburgess) [reply pkg/tsdb/cloudwatch/constants/metrics.go:452]: whoops, thank you added them back in

Comment (tristanburgess) [reply pkg/tsdb/cloudwatch/constants/metrics.go:452]: added InferenceComponents metrics as well

Comment (tristanburgess) [reply pkg/tsdb/cloudwatch/constants/metrics.go:452]: to be clear, for GroundTruth, the correct namespace is actually `AWS/SageMaker/LabelingJobs` and not `AWS/SageMaker`

**Review by iwysiu — APPROVED** [review]

**Pull request merged**